vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ImageServiceTest.java

```diff
@@ -61,4 +84,33 @@ public void test_getImageDimensions_fullHeight() {
         Assert.assertEquals(333, data.getWidth());
         Assert.assertEquals(200, data.getHeight());
     }
+    
+    @Test
+    public void test_getTotalPages_success() {
+        when(imageRepo.count()).thenReturn(12L);
+        assertEquals(2, serviceToTest.getTotalPages());
+        verify(imageRepo, times(2)).count();
```

Comment:
this isn't needed. we don't care how often count is called. If the implementation changes and count is assigned to a variable before doing the calculate, this test should still succeed as long as the result is correct.

Reply:
what about the case, when there are no images stored yet? there should be a test case, since this is a "special" situation.